core/src/main/java/brooklyn/BrooklynVersion.java

```diff
@@ -46,201 +58,333 @@
+     * <p/>
+     * In a packaged or library build of Brooklyn (normal usage) this should return false,
+     * and all OSGi components should be available.
+     * <p/>
+     * There is no longer any way to force this,
+     * such as the old BrooklynDevelopmentMode class;
+     * but that could easily be added if required (eg as a system property).
+     */
+    public static boolean isDevelopmentEnvironment() {
+        Boolean isDevEnv = IS_DEV_ENV.get();
+        if (isDevEnv != null) return isDevEnv;
+        synchronized (IS_DEV_ENV) {
+            isDevEnv = computeIsDevelopmentEnvironment();
+            IS_DEV_ENV.set(isDevEnv);
+            return isDevEnv;
+        }
+    }
+
+    private static boolean computeIsDevelopmentEnvironment() {
+        Enumeration<URL> paths;
+        try {
+            paths = BrooklynVersion.class.getClassLoader().getResources("brooklyn/BrooklynVersion.class");
+        } catch (IOException e) {
+            // shouldn't happen
+            throw Exceptions.propagate(e);
+        }
+        while (paths.hasMoreElements()) {
+            URL u = paths.nextElement();
+            // running fram a classes directory (including coverage-classes for cobertura) triggers dev env
+            if (u.getPath().endsWith("classes/brooklyn/BrooklynVersion.class")) {
+                try {
+                    log.debug("Brooklyn dev/src environment detected: BrooklynVersion class is at: " + u);
+                    return true;
+                } catch (Exception e) {
+                    Exceptions.propagateIfFatal(e);
+                    log.warn("Error reading manifest to determine whether this is a development environment: " + e, e);
+                }
+            }
+        }
+        return false;
+    }
+
+    public void logSummary() {
+        log.debug("Brooklyn version " + getVersion() + " (git SHA1 " + getSha1FromOsgiManifest() + ")");
+    }
+
+    /**
+     * @deprecated since 0.7.0, redundant with {@link #get()}
+     */
+    @Deprecated
+    public static String getVersionFromStatic() {
+        return VERSION_FROM_STATIC;
+    }
+
+    public static String get() {
+        return INSTANCE.getVersion();
+    }
+
+    /**
+     * @param mgmt The context to search for features.
+     * @return An iterable containing all features found in the management context's classpath and catalogue.
+     */
+    public static Iterable<BrooklynFeature> getFeatures(ManagementContext mgmt) {
+        Iterable<URL> manifests = ResourceUtils.create(mgmt).getResources(MANIFEST_PATH);
+
+        for (CatalogItem<?, ?> catalogItem : mgmt.getCatalog().getCatalogItems()) {
+            OsgiBrooklynClassLoadingContext osgiContext = new OsgiBrooklynClassLoadingContext(
+                    mgmt, catalogItem.getCatalogItemId(), catalogItem.getLibraries());
+            manifests = Iterables.concat(manifests, osgiContext.getResources(MANIFEST_PATH));
+        }
```

Comment:
I needed something similar the other day and used the following snipped:
```
        Maybe<OsgiManager> osgiManager = managementContext.getOsgiManager();
        if (osgiManager.isPresent()) {
            Framework framework = osgiManager.get().getFramework();
            Bundle[] bundles = framework.getBundleContext().getBundles();
            for (Bundle bundle : bundles) {
                if (bundle == framework) continue;
                if (Osgis.isExtensionBundle(bundle)) continue;

                try {
                    //TODO better .start the bundle on install
                    bundle.loadClass("Some.None.Existent.Class.To.Force.BundleWiring.Initialization");
                } catch (ClassNotFoundException e1) {}
                BundleWiring bundleWiring = bundle.adapt(BundleWiring.class);
                if (bundleWiring == null) continue; //bundle not resolved
                ClassLoader bundleClassLoader = bundleWiring.getClassLoader();
                //do something with the CL, i.e.
                bundleClassLoader.getResources(MANIFEST_PATH)
            }
        }
```

It's hackish and not necessarily better than your approach, best way is probably using OSGi services - but not feasible at this point.

One thing to watch out with scanning the catalog items is that the same bundle can be used in multiple catalog items, i.e. one jar providing multiple entities. Then you will get duplicates in the output.